agimus_demo_05_pick_and_place/agimus_demo_05_pick_and_place/hpp_client.py

```diff
@@ -0,0 +1,273 @@
+            srdfString += f'<disable_collisions link1="panda_link{i}_sc" link2="{self.manip_object.name}/base_link" reason="handled otherwise"/>'
+        srdfString += f'<disable_collisions link1="panda_hand_sc" link2="{self.manip_object.name}/base_link" reason="handled otherwise"/>'
+        srdfString += "</robot>"
+        self.robot.client.manipulation.robot.insertRobotSRDFModelFromString(
+            "panda", srdfString
+        )
+
+        # Create robot gripper handle, x-axis is facing down (for pregrasp)
+        self.ps.client.manipulation.robot.addGripper(
+            "panda/support_link",
+            "goal/gripper",
+            self.goal_obj_pose[:3] + [0, sqrt(2) / 2, 0, sqrt(2) / 2],
+            0.0,
+        )
+
+        # Lock gripper in open position.
+        # TODO: pass this as a parameter?
+        self.ps.createLockedJoint(
+            "locked_finger_1", "panda/panda_finger_joint1", [0.035]
+        )
+        self.ps.createLockedJoint(
+            "locked_finger_2", "panda/panda_finger_joint2", [0.035]
+        )
+        self.ps.setConstantRightHandSide("locked_finger_1", True)
+        self.ps.setConstantRightHandSide("locked_finger_2", True)
+        # Add handle of the objects
+        self.handles, self.goal_handles = get_obj_goal_handles(
+            prefix=self.manip_object.name + "/",
+            srdf_path=self.manip_object.srdfFilename,
+        )
+
+    def setup_problem(self):
+        newProblem()
+        self.set_robot()
+        self.set_problem()
+
+    def restart(self):
+        self.corba.reset_problem()
+        self.setup_problem()
+
+    def plan(self, q_init: list[float], q_goal: list[float] = None):
+        self.q_init = q_init + self.start_obj_pose + self.default_obstacle_pose
+        if q_goal is None:
+            q_goal = q_init.copy()
+        self.q_goal = q_goal + self.goal_obj_pose + self.default_obstacle_pose
+
+        self.binPicking = BinPicking(self.ps)
+        self.binPicking.objects = [self.manip_object.name, self.obstacle_object.name]
+        self.binPicking.robotGrippers = ["panda/panda_gripper"]
+        self.binPicking.goalGrippers = ["goal/gripper"]
+        self.binPicking.goalHandles = self.goal_handles
+        self.binPicking.handles = self.handles
+        self.binPicking.graphConstraints = ["locked_finger_1", "locked_finger_2"]
+
+        # TODO: restructure this
+        def disable_collision():
+            srdf_disable_collisions = """<robot>\n"""
+            srdf_disable_collisions_fmt = (
+                """  <disable_collisions link1="{}" link2="{}" reason=""/>\n"""
+            )
+            srdf_disable_collisions += srdf_disable_collisions_fmt.format(
+                "box/base_link", "part/base_link"
+            )
+            srdf_disable_collisions += "</robot>"
+            self.robot.client.manipulation.robot.insertRobotSRDFModelFromString(
+                "", srdf_disable_collisions
+            )
+
+        disable_collision()
```

Comment:
What is your intent here ?

Reply:
not sure, this is the code copied from Theo, I would say that this should be a part of .srdf file, not part of the code (but would need to think about it so I left it for now)

Reply:
Idem: same topic, same issue: https://github.com/agimus-project/agimus-demos/issues/59